frontend/src/components/pages/CampOverview/RemoveCamperModal/SelectModal.tsx

```diff
@@ -0,0 +1,117 @@
+import React from "react";
+
+import {
+  Box,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  Button,
+  Text,
+  Checkbox,
+  Stack,
+} from "@chakra-ui/react";
+
+import { Camper } from "../../../../types/CamperTypes";
+
+type SelectModalProps = {
+  camper: Camper;
+  campersToBeDeleted: Set<Camper>;
+  setCampersToBeDeleted: React.Dispatch<React.SetStateAction<Set<Camper>>>;
+  retrievedCampers: Camper[];
+  deselectAndClose: () => void;
+  setStatusAsConfirm: () => void;
+};
+
+const SelectModal = ({
+  camper,
+  campersToBeDeleted,
+  setCampersToBeDeleted,
+  retrievedCampers,
+  deselectAndClose,
+  setStatusAsConfirm,
+}: SelectModalProps): JSX.Element => {
+  const addCamperToBeDeleted = (newCamper: Camper) => {
+    setCampersToBeDeleted(new Set(campersToBeDeleted.add(newCamper)));
+  };
+
+  const removeCamperToBeDeleted = (oldCamper: Camper) => {
+    const newSet = new Set(campersToBeDeleted);
+    newSet.delete(oldCamper);
+    setCampersToBeDeleted(newSet);
+  };
+
+  const addAllCampers = () => {
+    const newSet = new Set<Camper>();
+
+    retrievedCampers.forEach((currentCamper: Camper) =>
+      newSet.add(currentCamper),
+    );
+
+    setCampersToBeDeleted(newSet);
+  };
+
+  return (
+    <ModalContent maxWidth="400px" maxHeight="80%">
+      <ModalHeader ml={8} mr={8} pt={8} pb={4} pl={0} pr={0}>
+        <Text textStyle="buttonSemiBold">Remove group members?</Text>
+      </ModalHeader>
+
+      <ModalBody pl={8} pr={8} pb={0} pt={0}>
+        <Box pt={4} pb={4}>
+          <Text textStyle="bodyRegular">
+            {camper.firstName} {camper.lastName} registered with the following
+            group members. Please select all the group members you would also
+            like to remove from this session:{" "}
```

Comment:
nit: change to "was registered for this camp session with the following campers. Please select all campers you would like to remove from this session"

(probably a bit redundant, but clearer about where we get these groups from)